mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb

```diff
@@ -7171,6 +7195,30 @@ resource "google_container_cluster" "primary" {
 `, cluster, np)
 }
 
+func testAccContainerCluster_withEnableKubernetesBetaAPIs(cluster, np string) string {
```

Comment:
I think this test is fragile and will fail in the near future. Is it okay to add this kind of test, or is there a better way to work around it?

Reply:
min_master_version would ideally be set using a `google_container_engine_versions` datasource similar to https://github.com/GoogleCloudPlatform/magic-modules/blob/85ff1c21e6aaff26aafa56c47d0af03ae6aa3726/mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb#L712 to avoid breaking when that version is no longer supported.

Is that your concern about fragility or is there something else going on?

Reply:
I apologize for the lack of clarity. This is a concern because this feature is only available since GKE 1.27. Therefore, we cannot use the `google_container_engine_versions` datasource in this case, correct? Another concern is that `authentication.k8s.io/v1beta1/selfsubjectreviews` will not be available in the near future. This is because the feature will be GAed and the v1beta1 API will be dropped (or this feature may be dropped before GA). I guess we need to maintain the available beta API feature in conjunction with the GKE version.

Reply:
GKE 1.27 is the latest available version on the STABLE release channel, so using the datasource shouldn't be an issue. I am not familiar with the internals of k8s versioning, so I'll need to look into that.

Reply:
Thanks! You're right. I changed to use datasource to specify `min_master_version`.

Reply:
Hi @toVersus thanks for creating this PR. the validation reads from new k8s releases open api spec to have list of valid beta apis available for enablement, so its in sync with upstream kubernetes. When the API is deprecated, this config is ignored and on upgrade the cluster's metadata will reflect that the api is no longer enabled. But clusters using deprecated apis are blocked from autoupgrade.